workflow/rules/bam.smk

```diff
@@ -0,0 +1,27 @@
+rule separate_bam_paired:
+    input:
+        "{bam_file}.bam",
+    output:
+        "{bam_file}.1.fq",
+        "{bam_file}.2.fq",
```

Comment:
For this and the other rule below, I would suggest to not write the fastqs into the same dir as the bam. Instead, write to a fixed path under results with sample and unit as wildcards. Also, store fq.gz (samtools can do this compression itself).